dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -87,9 +88,48 @@
 			ambiq,pwrcfg = <&pwrcfg 0x4 0x1000>;
 		};
 
-		pinctrl: pin-controller@40010000 {
-			compatible = "ambiq,apollo4-pinctrl";
+		gpio0: gpio@40010000 {
 			reg = <0x40010000 0x800>;
+
+			pinctrl: pin-controller {
+				compatible = "ambiq,apollo4-pinctrl";
+			};
```

Comment:
this looks weird, why is pinctrl a child? shouldn't it be the top node as in e.g. stm32?

Reply:
I changed it, so now the top node is `pinctrl`.